src/dotnet/CommandLine/CommandLineApplication.cs

```diff
@@ -334,7 +340,15 @@ public void ShowHelp(string commandName = null)
             for (var cmd = this; cmd != null; cmd = cmd.Parent)
             {
                 cmd.IsShowingInformation = true;
-                headerBuilder.Insert(6, string.Format(" {0}", cmd.Name));
+                if (cmd != this && cmd.Arguments.Any())
+                {
+                    var args = string.Join(" ", cmd.Arguments.Select(arg => arg.Name));
+                    headerBuilder.Insert(6, string.Format(" {0} {1}", cmd.Name, args));
```

Comment:
I don't think it matters that much - this is formatting code and most likely you don't need to know what this means, I'd rather not see the magic variable in a constant files in this case...